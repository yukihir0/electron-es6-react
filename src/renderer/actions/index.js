export function increment() {
  return { type: "INCREMENT" };
}
 
export function decrement() {
  return { type: "DECREMENT" };
}

export function asyncIncrement() {
  return dispatch => {
    setTimeout(() => {
      dispatch({type: "ASYNC_INCREMENT"});
    }, 1000);
  };
}

export function asyncDecrement() {
  return dispatch => {
    setTimeout(() => {
      dispatch({type: "ASYNC_DECREMENT"});
    }, 1000);
  };
}
