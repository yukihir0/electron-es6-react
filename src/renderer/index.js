import Hello from './components/hello.jsx'
