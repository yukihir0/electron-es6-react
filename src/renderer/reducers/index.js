const initState = {
  value: 0,
  async_value: 0,
};

export default (state = initState, action) => {
  switch (action.type) {
    case "INCREMENT":
      return Object.assign({}, state, {
        value: state.value + 1
      });
    case "DECREMENT":
      return Object.assign({}, state, {
          value: state.value - 1
      });
    case "ASYNC_INCREMENT":
      return Object.assign({}, state, {
        async_value: state.async_value + 1
      });
    case "ASYNC_DECREMENT":
      return Object.assign({}, state, {
          async_value: state.async_value - 1
      });
    default:
      return state;
  }
};
