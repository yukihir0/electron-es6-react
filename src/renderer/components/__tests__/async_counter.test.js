import React from "react";
import renderer from "react-test-renderer";
import { shallow, mount } from "enzyme";
import AsyncCounter from "../async_counter";

test("should render count", () => {
  const props = {
    async_value: 0
  };
  const element = shallow(
    <AsyncCounter {...props} />
  );

  expect(element.find("p").text()).toBe("Clicked: 0 times");
});

test("if increment button click, should call increment function", () => {
  const mockFn= jest.fn();
  const props = {
    async_value: 0,
    onAsyncIncrement: mockFn,
  };
  const element = shallow(
    <AsyncCounter {...props} />
  );

  element.find("#btn-increment").simulate("click");
  expect(mockFn).toHaveBeenCalled();
});

test("if decrement button click, should call decrement function", () => {
  const mockFn= jest.fn();
  const props = {
    async_value: 0,
    onAsyncDecrement: mockFn,
  };
  const element = shallow(
    <AsyncCounter {...props} />
  );

  element.find("#btn-decrement").simulate("click");
  expect(mockFn).toHaveBeenCalled();
});

test("snapshot", () => {
  const props = {
    async_value: 0
  };
  const element = renderer.create(<AsyncCounter {...props} />).toJSON();

  expect(element).toMatchSnapshot();
});
