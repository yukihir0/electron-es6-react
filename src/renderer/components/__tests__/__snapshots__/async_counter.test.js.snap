// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <p>
      <div
        className="col"
      >
        Clicked: 
        0
         times
      </div>
    </p>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <button
        className="col btn btn-outline-primary btn-block"
        id="btn-increment"
        type="button"
      >
        +
      </button>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <button
        className="col btn btn-outline-primary btn-block"
        id="btn-decrement"
        type="button"
      >
        -
      </button>
    </div>
  </div>
</div>
`;
